check_diff/tests/check_diff.rs

```diff
@@ -0,0 +1,80 @@
+use check_diff::{check_diff, compile_rustfmt, search_for_rs_files, CheckDiffError};
+use std::fs::File;
+use tempfile::Builder;
+
+#[test]
+fn search_for_files_correctly_non_nested() -> Result<(), Box<dyn std::error::Error>> {
+    let dir = Builder::new().tempdir_in("").unwrap();
+    let file_path = dir.path().join("test.rs");
+    let _tmp_file = File::create(file_path)?;
+
+    let iter = search_for_rs_files(dir.path());
+
+    let mut count = 0;
+    for _ in iter {
+        count += 1;
+    }
+
+    assert_eq!(count, 1);
+
+    Ok(())
+}
+
+#[test]
+fn search_for_files_correctly_nested() -> Result<(), Box<dyn std::error::Error>> {
+    let dir = Builder::new().tempdir_in("").unwrap();
+    let file_path = dir.path().join("test.rs");
+    let _tmp_file = File::create(file_path)?;
+
+    let nested_dir = Builder::new().tempdir_in(dir.path()).unwrap();
+    let nested_file_path = nested_dir.path().join("nested.rs");
+    let _ = File::create(nested_file_path)?;
+
+    let iter = search_for_rs_files(dir.path());
+
+    let mut count = 0;
+    for _ in iter {
+        count += 1;
+    }
+
+    assert_eq!(count, 2);
+
+    Ok(())
+}
+
+#[test]
+fn check_diff_test() -> Result<(), CheckDiffError> {
+    let tmp_dir = Builder::new().tempdir_in("").unwrap();
+    let runners = compile_rustfmt(
+        tmp_dir.path(),
+        "https://github.com/rust-lang/rustfmt".to_string(),
+        "rustfmt-1.4.32".to_string(),
+        None,
+    )?;
+
+    let dir = Builder::new().tempdir_in("").unwrap();
+    let file_path = dir.path().join("test.rs");
+    let _tmp_file = File::create(file_path)?;
+
+    let errors = check_diff(None, runners, dir.path());
+    assert_eq!(errors, 0);
+    Ok(())
+}
+
+#[test]
+fn format_simple_code() -> Result<(), CheckDiffError> {
+    let tmp_dir = Builder::new().tempdir_in("").unwrap();
+    let runners = compile_rustfmt(
+        tmp_dir.path(),
+        "https://github.com/rust-lang/rustfmt".to_string(),
+        "rustfmt-1.4.32".to_string(),
+        None,
+    )?;
```

Comment:
Instead of needing to compile both the `src` and `feature` binaries to test this can we simply just call `build_rustfmt_from_src` to build the local version of rustfmt? You may need to modify the function signature to take the current working directly so that we can set it for any of the subcommands with [Command::current_dir](https://doc.rust-lang.org/std/process/struct.Command.html#method.current_dir). You'll probably also need to add the current_working directory as an argument to `get_ld_library_path`, and update the call to `std::fs::copy("target/release/rustfmt", &binary_path)?;` to be relative to the current working directory.